docs/Client.md

```diff
@@ -0,0 +1,883 @@
+
+const server = createServer((request, response) => {
+  response.end('Hello, World!')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  client.request({
+    path: '/',
+    method: 'GET',
+  }).then(({ body }) => {
+    body.destroy()
+  }).catch(error => {
+    console.error(error) // should print an RequestAbortedError
+    client.close()
+    server.close()
+  })
+})
+```
+
+### `Client.stream(options, factory [, callback])`
+
+A faster version of `Client.request`. This method expects the second argument `factory` to return a [`Writable`](https://nodejs.org/api/stream.html#stream_class_stream_writable) stream which the response will be written to. This improves performance by avoiding creating an intermediate [`Readable`](https://nodejs.org/api/stream.html#stream_readable_streams) stream when the user expects to directly pipe the response body to a [`Writable`](https://nodejs.org/api/stream.html#stream_class_stream_writable) stream.
+
+As demonstrated in [Example 1 - Basic GET stream request](#example-1---basic-get-stream-request), it is strongly recommended to use the `option.opaque` property to avoid creating a closure for the `factory` method. This pattern works extremely well with Node.js Web Frameworks such as [Fastify](https://fastify.io). See [Example 2 - Stream to Fastify Response](#example-2---stream-to-fastify-response) for more details.
+
+Arguments:
+
+* **options** `RequestOptions`
+* **factory** `(data: StreamFactoryData) => stream.Writable`
+* **callback** `(error: Error | null, data: StreamData) => void` (optional) - If a `callback` is not provided, a `Promise` is returned
+
+Returns: `void | Promise<StreamData>` - Only returns a `Promise` if no `callback` argument was passed
+
+#### Parameter: `StreamFactoryData`
+
+* **statusCode** `number`
+* **headers** `http.IncomingHttpHeaders`
+* **opaque** `unknown`
+
+#### Parameter: `StreamData`
+
+* **opaque** `unknown`
+* **trailers** `Record<string, string>`
+
+#### Example 1 - Basic GET stream request
+
+```js
+'use strict'
+const { createServer } = require('http')
+const { Client } = require('undici')
+const stream = require('stream')
+
+const { PassThrough } = require('stream')
+
+const server = createServer((request, response) => {
+  response.end('Hello, World!')
+})
+
+server.listen(() => {
+  const client = new Client(`http://localhost:${server.address().port}`)
+
+  const pt = new PassThrough()
+  const bufs = []
+
+  client.stream({
+    path: '/',
+    method: 'GET',
+    opaque: { bufs, pt }
+  }, ({ statusCode, headers, opaque: { pt, bufs } }) => {
+    console.log(`response received ${statusCode}`)
+    console.log('headers', headers)
+    pt.on('data', buf => {
+      bufs.push(buf)
+    })
+    return pt
+  }).then(({ opaque: { bufs } }) => {
+
+    console.log(Buffer.concat(bufs).toString('utf8'))
+
+    client.close()
+    server.close()
+  }).catch(error => {
+    console.error(error)
+  })
+})
```

Comment:
This is example is a bit weird.

Reply:
Whats a better way to demonstrate a basic GET request for the stream method? Totally open to suggestions - my only requirement is that its a completely executable code block so users can copy-paste and run it themselves directly.

Reply:
just write it to a fs.createWriteStream?

Reply:
```js

({ opaque: filename, headers, statusCode }) => {
  // check status code

  // check headers for e.g. gzip compression

  return fs.createWriteStream(filename)
})
```

Reply:
or 

```js
let str = ''
({ opaque: filename, headers, statusCode }) => {
  // check status code

  // check headers for e.g. gzip compression

  return new Writable({
    write (chunk, encoding, callback) {
       str += chunk
    }
  })
})

```